service/src/main/kotlin/no/nav/su/se/bakover/service/regulering/ReguleringServiceImpl.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.su.se.bakover.service.regulering
+
+import arrow.core.Either
+import arrow.core.right
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.domain.regulering.BehandlingType
+import no.nav.su.se.bakover.domain.regulering.ReguleringRepo
+import no.nav.su.se.bakover.domain.regulering.VedtakSomKanReguleres
+import no.nav.su.se.bakover.domain.regulering.VedtakType
+import java.time.LocalDate
+
+class ReguleringServiceImpl(
+    private val reguleringRepo: ReguleringRepo,
+) : ReguleringService {
+
+    override fun hentAlleSakerSomKanReguleres(fraDato: LocalDate): Either<KanIkkeHenteSaker, SakerSomKanReguleres> {
+        return SakerSomKanReguleres(
+            saker = hentAlleSaker(fraDato),
+        ).right()
+    }
+
+    private fun hentVedtakSomKanReguleres(fomDato: LocalDate): List<VedtakSomKanReguleres> {
+        return reguleringRepo.hentVedtakSomKanReguleres(fomDato)
+    }
+
+    private fun hentAlleSaker(fraDato: LocalDate): List<SakSomKanReguleres> {
+        return hentVedtakSomKanReguleres(fraDato)
+            .filterNot { it.vedtakType == VedtakType.AVSLAG || it.vedtakType == VedtakType.AVVIST_KLAGE }
+            .groupBy { it.sakId }
+            .mapNotNull { (sakid, vedtakSomKanReguleres) ->
+                val minFra: LocalDate = maxOf(vedtakSomKanReguleres.minOf { it.fraOgMed }, fraDato)
+                val maxTil: LocalDate = vedtakSomKanReguleres.maxOf { it.tilOgMed }
+
+                val gjeldendeVedtakPrMnd = Periode.create(minFra, maxTil).tilMånedsperioder().map { mnd ->
+                    mnd to vedtakSomKanReguleres.filter {
+                        val vedtaksperiode = Periode.create(it.fraOgMed, it.tilOgMed)
+                        vedtaksperiode.inneholder(mnd)
+                    }.maxByOrNull { it.opprettet.instant }!!
+                }.filterNot { it.second.vedtakType == VedtakType.OPPHØR }.ifEmpty {
+                    return@mapNotNull null
+                }
+
+                val type =
+                    if (gjeldendeVedtakPrMnd.any { it.second.behandlingType == BehandlingType.MANUELL || it.second.vedtakType == VedtakType.STANS_AV_YTELSE }) {
```

Comment:
Jeg ser det ikke er tatt hensyn til INGEN_ENDRING, som er helt greit i dagens situasjon. Siden vi ikke har noen INGEN_ENDRING i produksjon. Per tidspunkt kan en g-regulering føre til INGEN_ENDRING, men det snakket akkurat Jacob og jeg om at vi bør ikke støtte tilfeller der en regulering fører til INGEN_ENDRING eller en negativ differanse i utmålinga av stønaden.

Men det vil putte litt mer press på oss til å gjøre den endringen jeg disuterer over.

Reply:
@hestad Er vel to hensyn rundt "ingen endring". 
1) Dersom det hadde eksistert et Ingen-endring-vedtak så skal det reguleres som alle andre vedtak, og vil da ta med de små endringene i beløp som er gjort i tillegg til selve reguleringen.
2) Det skal ikke være mulig å havne i "ingen endring" eller "opphør" etter reguleringen, det skal kun være en beløpsendring. I teorien kan kanskje noen havne på 0-utbetalinger? Må evt. høre litt med fag hva de tenker om den situasjonen, men den er nok ikke så sannsynlig i praksis, siden både sats og fradrag burde endre seg i tilsvarende prosent, og Grunnbeløpet har ikke blitt senket i overskuelig historie.